modules/security-best-practices/pages/rust-canister-development-security-best-practices.adoc

```diff
@@ -10,19 +10,20 @@ The controller of a canister can change / update the canister whenever they like
 
 ==== Recommendation
 
-* To implement an immutable canister smart contract, you cannot be the controller of the canister. Control must be passed to the service nervous system (SNS) or a decentralized governance system.
-* Another option to create an immutable canister smart contract is to remove the canister controller completely. But note that this implies that the canister cannot be upgraded, which may have severe implications in case e.g. a bug would be found.
-* Note that, contrary to some other blockchains, also immutable smart contracts need cycles to run, and they can receive cycles.
+* Consider passing canister control to a decentralized governance system such as the Internet Computer's Service Nervous System (SNS), so that changes to the canister must be proposed and voted upon. If the SNS is used, use an NNS-blessed version running on the SNS subnet, meaning that SNS is under NNS control. The SNS will be available soon.
+* Another option would be to create an immutable canister smart contract by removing the canister controller completely. However, note that this implies that the canister cannot be upgraded, which may have severe implications in case e.g. a bug were found. The option to use a decentralized governance system and thus being able to upgrade smart contracts is a big advantage of the Internet Computer ecosystem.
+* It is also possible to implement a DAO (link:https://en.wikipedia.org/wiki/Decentralized_autonomous_organization[Decentralized Autonomous Organization]) on the IC from scratch. If you decide to do this (e.g. along the lines of the link:https://smartcontracts.org/docs/samples/dao.html[basic DAO example]), be aware that this is security critical and must be security reviewed carefully.
+* Note that, contrary to some other blockchains, smart contracts (also immutable ones) need cycles to run, and they can receive cycles.
```

Comment:
Immutable canisters are related because the presumed goal of the developer is that _there is no centralized control_.

Reply:
right I see that the subject is related, but this bullet point seems to raise a point that people could be suspicious for _for immutable canisters_. If so, I think it should be a subpoint of the immutable canisters mentioned rather than another point.

Or stated differently: it is unclear to me why this information is in a bullet list of things that seems to list options for having non-central controllers for a canister (it does not add another option for this). And since I am confused about that I suspect that others might too. Therefore, I wonder if the reason can be made explicit or if it makes sense to move this bullet point to somewhere else

Reply:
Thanks. Good point, this does not fit the structure, as the other points list options for creating smart contracts. This is a purely informational (and not really essential) point that relates to the immutable contracts. I added it as a sub point there, does that work for you? If you have a better idea on where to place this, please let me know.